### PR TITLE
Introduce AL2023 into the CI test matrix

### PR DESCRIPTION
**Pull request by pb8o**
## Changes

...

## Reason

...

## License Acceptance

By submitting this pull request, I confirm that my contribution is made under
the terms of the Apache 2.0 license. For more information on following
Developer Certificate of Origin and signing off your commits, please check
[`CONTRIBUTING.md`][3].

## PR Checklist

- [ ] If a specific issue led to this PR, this PR closes the issue.
- [x] The description of changes is clear and encompassing.
- [x] Any required documentation changes (code and docs) are included in this PR.
- [ ] API changes follow the [Runbook for Firecracker API changes][2].
- [ ] User-facing changes are mentioned in `CHANGELOG.md`.
- [x] All added/changed functionality is tested.
- [ ] New `TODO`s link to an issue.
- [x] Commits meet [contribution quality standards](https://github.com/firecracker-microvm/firecracker/blob/main/CONTRIBUTING.md#contribution-quality-standards).

---

- [x] This functionality cannot be added in [`rust-vmm`][1].

[1]: https://github.com/rust-vmm
[2]: https://github.com/firecracker-microvm/firecracker/blob/main/docs/api-change-runbook.md
[3]: https://github.com/firecracker-microvm/firecracker/blob/main/CONTRIBUTING.md

### CODE REVIEW

File: tests/framework/utils_vsock.py [hunk 1]
```diff
@@ -235,12 +235,13 @@ def check_guest_connections(vm, server_port_path, blob_path, blob_hash):
...
     cmd += "for w in $workers; do wait $w || exit -1; done"
 
-    ecode, _, _ = vm.ssh.run(cmd)
+    ecode, stdout, stderr = vm.ssh.run(cmd)
 
     echo_server.exit()
     assert echo_server.error is None
 
-    assert ecode == 0, ecode
+    print(stdout.read())
```
Comment (roypat) [inline tests/framework/utils_vsock.py:183]: is this print intended here?

also, probably out of scope here, but there's this annoying inconsistency between `utils.run_cmd` and `ssh.run` where the first returns strings and the second returns these objects that require .read().  :(

File: tools/devctr/pyproject.toml [hunk 2]
```diff
@@ -18,6 +18,7 @@ psutil = "^5.9.5"
 pylint = "^2.17.4"
 pytest = "^7.3.1"
 pytest-json-report = "^1.5.0"
+pytest-rerunfailures = "^11.1.2"
```
Comment (roypat) [inline tools/devctr/pyproject.toml:21]: thanks!

**Review by roypat — APPROVED** [review]

File: tests/framework/utils_vsock.py (hunk 1 above)
Comment (pb8o) [reply tests/framework/utils_vsock.py:183]: yep, the `read()` thing I already have it in another patch series. Having a similar interface for those objects would be good too.

Comment (roypat) [reply tests/framework/utils_vsock.py:183]: o, whoop! :D

**Review by ShadowCurse — APPROVED** [review]

**Pull request merged**